services/UserService.js

```diff
@@ -28,4 +28,20 @@ export default class UserService extends BaseService {
       throw new Error(error);
     }
   }
+
+  async softDelete(id) {
+    try {
+      const response = await this.model.update(
+        { isDeleted: true },
+        {
+          where: {
+            id,
+          },
+        },
+      );
+      return response;
+    } catch (error) {
+      throw new Error(error);
+    }
+  }
```

Comment:
The soft delete should remove entry from the UserGroup table too as required in the assignment.

Reply:
If any record gets removed from the DB, then all linked records should be removed from UserGroup as well.